v3/src/models/data/formula-types.ts

```diff
@@ -15,4 +27,23 @@ export interface IGlobalValueDependency {
   globalId: string
 }
 
-export type IFormulaDependency = ILocalAttributeDependency | IGlobalValueDependency
+export interface ILookupDependency {
+  type: "lookup"
+  dataSetId: string
+  attrId: string
+  keyAttrId?: string
+}
+
+export type IFormulaDependency = ILocalAttributeDependency | IGlobalValueDependency | ILookupDependency
+
+export type EvalValue = string | number | boolean
+
+export interface IFormulaMathjsFunction {
+  rawArgs: boolean
+  isAggregate: boolean
+  evaluate: (args: MathNode[], mathjs: any, scope: FormulaMathJsScope) => EvalValue | EvalValue[]
+  canonicalize?: (args: MathNode[], displayNameMap: DisplayNameMap) => void
+  getDependency?: (args: MathNode[]) => IFormulaDependency
+}
+
+export type ICODAPMathjsFunctionRegistry = Record<string, IFormulaMathjsFunction>
```

Comment:
```suggestion
export type CODAPMathjsFunctionRegistry = Record<string, IFormulaMathjsFunction>
```
For simple types (rather than interfaces) I'd leave off the `I`, but that's a personal preference.

Reply:
I also usually follow this naming. Fixed in the last comment.